### PR TITLE
remove observation from next_word_prob

### PR DESCRIPTION
**Pull request by alexholdenmiller**
also removed the caching from seq2seq since I was having a bug with it after these changes, could add back in later

### CODE REVIEW

**Comment by jaseweston** [comment]
oh is this gonna slow it down a lot? that wouldn't be good

File: projects/convai2/baselines/seq2seq/eval_ppl.py [hunk 1]
```diff
@@ -43,23 +45,14 @@ def next_word_probability(self, observation, partial_out):
         e.g.
         {'text': 'Run test program.'}, ['hello'] => {'world': 1.0}
         """
-        if not hasattr(self, 'prev_enc'):
-            self.prev_enc = None
-            self.last_text = None
-        if observation['text'] != self.last_text:
```
Comment (jaseweston) [inline projects/convai2/baselines/seq2seq/eval_ppl.py]: is it a big speed difference? you could definitely keep this by simply storing the last text as before, i dont see the difference?

Comment (alexholdenmiller) [reply projects/convai2/baselines/seq2seq/eval_ppl.py]: fixed!

**Comment by alexholdenmiller** [comment]
there may have been one exchange where the person said the same thing twice in a row and that broke it, but regardless I found a safe way to calculate whether it should reset the cache

**Pull request merged**